api/src/main/java/org/pmiops/workbench/mail/MailServiceImpl.java

```diff
@@ -1,16 +1,118 @@
 import javax.mail.Transport;
 
 @Service
 public class MailServiceImpl implements MailService {
 
-    public void send(Message msg) throws MessagingException {
-        Transport.send(msg);
+  private final Provider<MandrillApi> mandrillApiProvider;
+  private final Provider<CloudStorageService> cloudStorageServiceProvider;
+  private Provider<WorkbenchConfig> workbenchConfigProvider;
+  private static final Logger log = Logger.getLogger(MailServiceImpl.class.getName());
+
+  enum Status {REJECTED, API_ERROR, SUCCESSFUL}
+
+  @Autowired
+  public MailServiceImpl(Provider<MandrillApi> mandrillApiProvider,
+                         Provider<CloudStorageService> cloudStorageServiceProvider,
+                         Provider<WorkbenchConfig> workbenchConfigProvider) {
+    this.mandrillApiProvider = mandrillApiProvider;
+    this.cloudStorageServiceProvider = cloudStorageServiceProvider;
+    this.workbenchConfigProvider = workbenchConfigProvider;
+  }
+
+  public void send(Message msg) throws MessagingException {
+    Transport.send(msg);
+  }
+
+  public void sendWelcomeEmail(String contactEmail, String password, User user) throws MessagingException {
+    String apiKey = cloudStorageServiceProvider.get().readMandrillApiKey();
+    int retries = workbenchConfigProvider.get().mandrill.sendRetries;
+    MandrillApiKeyAndMessage keyAndMessage = new MandrillApiKeyAndMessage();
+    keyAndMessage.setKey(apiKey);
+    MandrillMessage msg = buildWelcomeMessage(contactEmail, password, user);
+    keyAndMessage.setMessage(msg);
+    do {
+      retries--;
+      ImmutablePair attempt = trySend(keyAndMessage);
+      Status status = Status.valueOf(attempt.getLeft().toString());
+//            Status status = Status.valueOf(attempt.getLeft().toString());
+      switch (status) {
+        case API_ERROR:
+          log.log(Level.WARNING, String.format(
+            "ApiException: Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, user.getName(), attempt.getRight().toString()));
+          if (retries == 0) {
+            log.log(Level.SEVERE, String.format(
+              "ApiException: On Last Attempt! Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, user.getName(), attempt.getRight().toString()));
+            throw new MessagingException("Sending email failed: " + attempt.getRight().toString());
+          }
+          break;
+
+        case REJECTED:
+          log.log(Level.SEVERE, String.format(
+            "Messaging Exception: Welcome Email to '%s' for user '%s' not sent: %s", contactEmail, user.getName(), attempt.getRight().toString()));
+          throw new MessagingException("Sending email failed: " + attempt.getRight().toString());
+
+        case SUCCESSFUL:
+          log.log(Level.INFO, String.format(
+            "Welcome Email to '%s' for user '%s' was sent.", contactEmail, user.getName()));
+          return;
+
+        default:
+          log.log(Level.SEVERE, String.format(
```

Comment:
I could have sworn I added a comment on this, but it seems to be lost. I think this needs to check the retries count and only log and fail if the count is at zero.